### PR TITLE
Add findf2py

### PR DESCRIPTION
**Pull request by xoviat**

### CODE REVIEW

**Comment by xoviat** [comment]
Ping @jcfr

File: skbuild/resources/cmake/FindF2py.cmake [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Find the f2py program, which comes as part of NumPy.
```
Comment (scopatz) [inline skbuild/resources/cmake/FindF2py.cmake]: This originally was part of the PyNE package.  Please include the copyright statement, as per the BSD license. Thanks!

Comment (xoviat) [reply skbuild/resources/cmake/FindF2py.cmake]: Done.

Comment (scopatz) [reply skbuild/resources/cmake/FindF2py.cmake]: Thanks so much @xoviat!

**Comment by jcfr** [comment]
Thanks for the contribution @xoviat 

Would you mind:
* updating the comment so that it include comment like `#.rst:` similar to what is done in the other modules ? 
* adding an entry into https://github.com/scikit-build/scikit-build/blob/master/docs/cmake-modules.rst

You can then locally check the generated documentation doing:

```
cd scikit-build
pip install -r requirements-docs.txt
make docs
```

**Comment by jcfr** [comment]
travis-ci build failure should be addressed in #275

**Comment by xoviat** [comment]
Ping @jcfr

File: docs/cmake-modules.rst [hunk 2]
```diff
@@ -21,6 +22,7 @@ They can be included using ``find_package``:
     find_package(Cython REQUIRED)
     find_package(NumPy REQUIRED)
     find_package(PythonExtensions REQUIRED)
+    find_package(F2PY REQUIRED)
```
Comment (jcfr) [inline docs/cmake-modules.rst]:
```
find_package(F2py REQUIRED)
```

File: skbuild/resources/cmake/FindF2py.cmake [hunk 3]
```diff
@@ -0,0 +1,98 @@
+#.rst:
+#
+# This module defines CMake functions to build Python extension modules and
+# stand-alone executables.
```
Comment (jcfr) [inline skbuild/resources/cmake/FindF2py.cmake]: Look  like the description should be inspired from text found at http://www.f2py.com/

File: docs/cmake-modules.rst (hunk 2 above)
Comment (xoviat) [reply docs/cmake-modules.rst]: Not the right fix. I renamed the module.

File: skbuild/resources/cmake/FindF2PY.cmake [hunk 4]
```diff
@@ -0,0 +1,98 @@
+#.rst:
+#
+# This module defines CMake functions to build Python extension modules and
+# stand-alone executables.
```
Comment (jcfr) [inline skbuild/resources/cmake/FindF2PY.cmake]: Would it make sense to tweak the description based on text found at http://www.f2py.com/  ?

File: skbuild/resources/cmake/FindF2PY.cmake [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      set(F2PY_EXECUTABLE ${F2PY_EXECUTABLE3})
+    endif(F2PY_EXECUTABLE3)
+  elseif(${PYTHON_VERSION_MAJOR} LESS 3)
+    # because arch is dumb
+    find_program(F2PY_EXECUTABLE2 F2PY2 ${DEPS_BIN_HINTS})
+    if(F2PY_EXECUTABLE2)
+      set(F2PY_EXECUTABLE ${F2PY_EXECUTABLE2})
+    endif(F2PY_EXECUTABLE2)
+  endif(${PYTHON_VERSION_MAJOR} GREATER 2)
+endif(NOT F2PY_EXECUTABLE)
```
Comment (jcfr) [inline skbuild/resources/cmake/FindF2PY.cmake]: Consider running this script to remove block end arguments:

https://github.com/jcfr/cmake-auto-style/blob/master/docs/cmake-commands-remove-block-end-arguments.md#cmake-commands-remove-block-end-arguments

File: skbuild/resources/cmake/FindF2PY.cmake [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+if(F2PY_EXECUTABLE)
+  SET(F2PY_FOUND TRUE)
```
Comment (jcfr) [inline skbuild/resources/cmake/FindF2PY.cmake]: `SET` -> `set`

File: skbuild/resources/cmake/FindF2PY.cmake [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+  else(F2PY_VERSION_N GREATER 1)
+    set(F2PY_VERSION_MINOR 0)
+  endif(F2PY_VERSION_N GREATER 1)
+  if(F2PY_VERSION_N GREATER 2)
+    list(GET F2PY_VERSION_LIST 2 F2PY_VERSION_MICRO)
+  else(F2PY_VERSION_N GREATER 2)
+    set(F2PY_VERSION_MICRO 0)
+  endif(F2PY_VERSION_N GREATER 2)
+else()
+  SET(F2PY_FOUND FALSE)
```
Comment (jcfr) [inline skbuild/resources/cmake/FindF2PY.cmake]: `SET` -> `set`

**Comment by jcfr** [comment]
Thanks @xoviat , after adding an entry to https://github.com/scikit-build/scikit-build/blob/master/CHANGES.rst#next-release, we should be all set.

**Comment by xoviat** [comment]
@jcfr Done.

**Comment by jcfr** [comment]
@xoviat  I rebased, squashed the commits and improved the documentation.

Could you review the "Example Usage" section I added ?  (  I inspired from  code found here: https://github.com/pyne/pyne/blob/master/pyne/CMakeLists.txt)

**Comment by xoviat** [comment]
So the idea here is actually to eliminate `numpy.distutils` entirely over time and replace it with scikit-build. We can't do that if we write `f2py -c` because `numpy.distutils` is still used. Instead we want to use `f2py` to generate the C files and then link them to the Python extension.

I'm also not happy with the way that Cython support was implemented in scikit-build. Ideally, you would be able to write the following and have scikit-build autogenerate the cython and f2py targets:

```
add_library(LIBRARY_NAME
    input.pyf.src
    input.pyx
) 
```

If you look at [what we're replacing](https://github.com/scipy/scipy/blob/5531da38cec429f3cf2f9aca0f698e2130562592/scipy/fftpack/setup.py), we essentially need to translate the `add_library` and `add_extension` function calls on a one-to-one basis.

**Comment by xoviat** [comment]
@jcfr Can you merge this?

**Comment by jcfr** [comment]
Closing. This is superseded by https://github.com/scikit-build/scikit-build/pull/276

**Pull request closed without merging**